engines/freescape/freescape.cpp

```diff
@@ -0,0 +1,609 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "common/config-manager.h"
+#include "common/events.h"
+#include "common/file.h"
+#include "common/math.h"
+#include "common/unzip.h"
+#include "graphics/cursorman.h"
+
+#include "freescape/freescape.h"
+#include "freescape/language/8bitDetokeniser.h"
+#include "freescape/neo.h"
+
+namespace Freescape {
+
+FreescapeEngine *g_freescape = NULL;
+
+FreescapeEngine::FreescapeEngine(OSystem *syst, const ADGameDescription *gd)
+	: Engine(syst), _gameDescription(gd), _gfx(nullptr) {
+	g_freescape = this;
+	if (!ConfMan.hasKey("render_mode") || ConfMan.get("render_mode").empty())
+		_renderMode = "ega";
+	else
+		_renderMode = ConfMan.get("render_mode");
+
+	_binaryBits = 0;
+	_screenW = 320;
+	_screenH = 200;
+
+	if (isAmiga()) {
+		_renderMode = "amiga";
+	} else if (isAtariST()) {
+		_renderMode = "atari";
+	}
+
+	if (gd->extra)
+		_variant = gd->extra;
+	else
+		_variant = "FullGame";
```

Comment:
Normally, we use gameflags (GF_xxx) for this.

Reply:
Fair enough. I'm using this:
```c++
enum DrillerReleaseFlags {
		ADGF_AMIGA_RETAIL = (1 << 0),
		ADGF_AMIGA_BUDGET = (1 << 1),
};
```